test/projectile-test.el

```diff
@@ -861,3 +861,30 @@ test temp directory"
     (spy-on 'projectile-project-type :and-return-value "bar")
     (let ((projectile-mode-line-prefix " Pro"))
       (expect (projectile-default-mode-line) :to-equal " Pro[foo:bar]"))))
+
+(describe "projectile--directory-p"
+  (it "tests which directory exists"
+    (expect (projectile--directory-p nil) :to-be nil)
+    (expect (projectile--directory-p "asdf") :to-be nil)
+    (expect (projectile--directory-p user-emacs-directory) :to-be-truthy)))
+
+(describe "projectile-dir-files"
+  :var (indexed-dir)
+  (it "list project files in project directories, otherwise fails"
+    (setq indexed-dir "a/")
+    (spy-on 'file-directory-p :and-call-fake
+            (lambda (filename) (equal filename "/my/root/")))
+    (spy-on 'projectile-patterns-to-ignore :and-return-value nil)
+    (spy-on 'projectile-index-directory :and-call-fake
+            (lambda (dir patterns progress-reporter)
+              (setq indexed-dir dir)
+              '("/my/root/a/b/c" "/my/root/a/d/e")))
+    (spy-on 'projectile-dir-files-alien :and-return-value '("a/b/c" "a/d/e"))
+    (spy-on 'cd :and-return-value nil)
+    (expect (projectile-dir-files "/my/root/") :to-equal '("a/b/c" "a/d/e"))
```

Comment:
TODO: call with `projectile-indexing-method` equal to multiple values like in the original test